llama_index/embeddings/clip.py

```diff
@@ -0,0 +1,136 @@
+import logging
+from typing import Any, List
+
+from llama_index.bridge.pydantic import Field, PrivateAttr
+from llama_index.embeddings.base import (
+    DEFAULT_EMBED_BATCH_SIZE,
+    BaseEmbedding,
+    Embedding,
+)
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_CLIP_MODEL = "ViT-B/32"
+
+
+class ClipEmbedding(BaseEmbedding):
```

Comment:
related to @logan-markewich 's comment. 

Should we define a new interface for multimodal embeddings (e.g.`BaseMultimodalEmbedding`)? It should e.g. enforce the subclass to implement  `get_image_embeddings` in addition to the get text / query embeddings.